modules/users/server/config/strategies/facebook.js

```diff
@@ -36,6 +37,18 @@ module.exports = function (config) {
 
       // Save the user OAuth profile
       users.saveOAuthUserProfile(req, providerUserProfile, done);
+
+      function generateUsername(profile) {
+        var username = '';
+
+        if (profile.emails) {
+          username = profile.emails[0].value.split('@')[0];
+        } else if (profile.name) {
+          username = profile.name.givenName[0] + profile.name.familyName;
+        }
+
+        return username.toLowerCase() || undefined;
```

Comment:
I saw this, but it does not handle it as nicely.  If there is no username, the app will not create a user, takes you to the login page.  Email is not required and if you reject access to email, then the current implementation does not work.  The case above is much more forgiving.  profile.name will always be returned, because that is the absolute minimum when authenticating with facebook.